Update the Ballerina OpenAPI connector generation guide (#9898)
ballerina-platform/ballerina-dev-website: swan-lake/resources/featured-scenarios/create-your-first-connector-with-ballerina.md

Detailed information on the Ballerina connector structure can be found in the [Ballerina module contribution guide](https://github.com/ballerina-platform/ballerina-library/blob/main/docs/adding-a-new-ballerina-module.md#directory-structure).

## Step 2: Prepare the OpenAPI definition

1. Find the OpenAPI definition for the API you want to create a connector. This is usually available in the API documentation.
   Example: For Twitter, you can get the latest API definition from the [Twitter OpenAPI endpoint](https://api.twitter.com/2/openapi.json).

2. Save the file as `openapi.yaml` (or `openapi.json`) in the `docs/spec` directory of your project.

3. To improve compatibility and readability before generating the Ballerina client, run the following preprocessing steps using the Ballerina OpenAPI tool:

   **a. Flatten the OpenAPI definition**

   This step relocates all inline embedded schemas to the `components` section to improve readability and reduce redundancy.
   

>    $ bal openapi flatten -i docs/spec/openapi.yaml -o docs/spec

Reviewer: ```suggestion
```
   $ bal openapi flatten -i docs/spec/openapi.yaml -o docs/spec
```

Detailed information on the Ballerina connector structure can be found in the [Ballerina module contribution guide](https://github.com/ballerina-platform/ballerina-library/blob/main/docs/adding-a-new-ballerina-module.md#directory-structure).

## Step 2: Prepare the OpenAPI definition

1. Find the OpenAPI definition for the API you want to create a connector. This is usually available in the API documentation.
   Example: For Twitter, you can get the latest API definition from the [Twitter OpenAPI endpoint](https://api.twitter.com/2/openapi.json).

2. Save the file as `openapi.yaml` (or `openapi.json`) in the `docs/spec` directory of your project.

3. To improve compatibility and readability before generating the Ballerina client, run the following preprocessing steps using the Ballerina OpenAPI tool:

   **a. Flatten the OpenAPI definition**

   This step relocates all inline embedded schemas to the `components` section to improve readability and reduce redundancy.
   

>    ```bash

Reviewer: Remove bash
```suggestion
   ```
```

>    bal openapi flatten -i docs/spec/openapi.yaml -o docs/spec

Reviewer: Add `$` if you want the user to be able to copy these commands.
```suggestion
   $ bal openapi flatten -i docs/spec/openapi.yaml -o docs/spec
```

   ```
   
   This command will generate a `flattened_openapi.yaml` file in the `docs/spec` directory.

   **b. Align the flattened OpenAPI definition**

   This aligns the OpenAPI definition according to the best practices of Ballerina..
   
   ```bash
   bal openapi align -i docs/spec/flattened_openapi.yaml -o docs/spec
   ```

>    This command will generate a `aligned_ballerina_openapi.yaml` file in the `docs/spec` directory.

Reviewer 2: ```suggestion
   This command will generate a file named `aligned_ballerina_openapi.yaml` in the `docs/spec` directory.
```

2. Save the file as `openapi.yaml` (or `openapi.json`) in the `docs/spec` directory of your project.

3. To improve compatibility and readability before generating the Ballerina client, run the following preprocessing steps using the Ballerina OpenAPI tool:

>    > **Note:** These preprocessing steps often reduce the need for manual specification sanitization. However, if further changes are needed (e.g. modifying security schemes or redefining schemas), document them in `docs/spec/sanitations.md`.

Reviewer 3: Shall we move it to the last Note as a point?

>    **a. Flatten the OpenAPI spec**

Reviewer 3: ```suggestion
   **a. Flatten the OpenAPI specification**
```

   This step relocates all inline embedded schemas to the `components` section to improve readability and reduce redundancy.
   
   ```bash
   bal openapi flatten -i docs/spec/openapi.yaml -o docs/spec
   ```
   
   This command will generate a `flattened_openapi.yaml` file in the `docs/spec` directory.

   **b. Align the flattened spec**

>    This aligns the OpenAPI file with Ballerina’s naming conventions by adding `x-ballerina-name` extensions where needed.

Reviewer 3: We are now supporting three alignments:
- `name`: Align according to the best naming practices of Ballerina. The Ballerina name extensions(`x-ballerina-name`) are added to the schemas which can not be modified directly.
- `doc`: Align according to the best documentation practices of Ballerina.
- `basepath`: Extract the common base path from the operations and add it to the
   server url.

Reviewer 3: We need to update this section as well with the new alignments: https://ballerina.io/learn/openapi-tool/#modify-the-openapi-contract-using-ballerina-preferred-naming-conventions

Author: Do we need to include thwese in here as well? I will update the OpenAPI learn page in a separate PR later..

Reviewer 3: We are mentioning about the naming convention right? Better to have a common statement here:
```suggestion
   This aligns the OpenAPI specification according to the best practices of Ballerina..
```

Author: ack
